lib/active_merchant/billing/gateways/d_local.rb

```diff
@@ -112,7 +112,8 @@ def add_additional_data(post, options)
       def add_country(post, card, options)
         return unless address = options[:billing_address] || options[:address]
```

Comment:
We need to add a provision for `options[:country]` in this or statement, otherwise if it's passed through options but the address hashes empty, we'll return of the method

Reply:
ah right thanks for catching that!